firedrake/adjoint_utils/blocks/solving.py

```diff
@@ -630,6 +637,53 @@ def _forward_solve(self, lhs, rhs, func, bcs, **kwargs):
         func.assign(self._ad_nlvs._problem.u)
         return func
 
+    def _adjoint_solve(self, dJdu, compute_bdy):
+        dJdu_copy = dJdu.copy()
+        # Homogenize and apply boundary conditions on adj_dFdu and dJdu.
+        bcs = self._homogenize_bcs()
+        for bc in bcs:
+            bc.apply(dJdu)
+
+        # Update the left hand side coefficients of the adjoint equation.
+        problem = self._ad_adj_solver._problem
+        for block_variable in self.get_dependencies():
+            # The self.adj_F coefficients hold the forward output
+            # references.
+            if block_variable.output in self.adj_F.coefficients():
```

Comment:
This condition is required. We are looping over self._dependencies that are not necessarily a self.adj_F coefficient.